core/src/main/java/org/mule/runtime/core/util/AttributeEvaluator.java

```diff
@@ -42,6 +42,14 @@ public AttributeEvaluator initialize(final ExtendedExpressionManager expressionM
     return this;
   }
 
+  private String sanitize(String attributeValue) {
```

Comment:
because the expression can be in a CDATA, it means it can start or end with `\n`, `\t`, spaces and things of the sort....  This removes that and makes it so that the validaiton regex recognizes this as an expression.

What was discussed with Ana is that this should move to the ExpressionLanguage, but at the time of sending this PR that was still not finished up.